collect_app/src/main/java/org/odk/collect/android/formentry/saving/FormSaveViewModel.java

```diff
@@ -80,6 +87,36 @@ public void saveForm(Uri instanceContentURI, boolean shouldFinalize, String upda
         }
     }
 
+    // Cleanup when user exits a form without saving
+    public void removeTempInstance() {
+
+        if (formController != null && formController.getInstanceFile() != null) {
+            SaveFormToDisk.removeSavepointFiles(formController.getInstanceFile().getName());
+
+            // if it's not already saved, erase everything
+            if (!InstancesDaoHelper.isInstanceAvailable(getAbsoluteInstancePath())) {
+                // delete media first
+                String instanceFolder = formController.getInstanceFile().getParent();
+                Timber.i("Attempting to delete: %s", instanceFolder);
+                File file = formController.getInstanceFile().getParentFile();
+                int images = MediaUtils.deleteImagesInFolderFromMediaProvider(file);
+                int audio = MediaUtils.deleteAudioInFolderFromMediaProvider(file);
+                int video = MediaUtils.deleteVideoInFolderFromMediaProvider(file);
+
+                Timber.i("Removed from content providers: %d image files, %d audio files and %d audio files.",
+                        images, audio, video);
+                FileUtils.purgeMediaPath(instanceFolder);
+            }
+        } else {
+            Timber.w("null returned by getFormController()");
```

Comment:
We can probably remove this logging now as we should have a clearer picture whether the `FormController` is null or not here.